STAR-Request/draft-sheffer-acme-star-delegation.md

```diff
@@ -0,0 +1,302 @@
+
+~~~
+   {
+     "status": "valid",
+     "expires": "2016-01-01T00:00:00Z",
+
+     "identifiers": [
+      {
+        "type": "dns",
+        "value": "abc.ndc.dno.example",
+        "delegated": true,
+        "cname": "abc.ndc.example"
+      }
+     ],
+
+     "authorizations": [],
+
+     "finalize": "https://acme.dno.example/acme/order/TOlocE8rfgo/finalize",
+
+     "certificate": "https://acme.ca.example/acme/order/yTyr23sSDg9"
+   }
+~~~
+
+Note that at this point in the flow, the IdO can add the CNAME records to its zone.
+
+### Order Object on the IdO-CA side
+
+The Order from the IdO to the ACME CA SHOULD strip the "cname" attribute from the twin Order from the NDC ({{sec-profile-ndc-to-ido}}), and MUST contain the necessary STAR extensions.  In addition, in order to allow the NDC to download the certificate using the unauthenticated GET, it MUST contain the recurrent-certificate-get attribute and set its value to true.
+
+### Capability Discovery
+
+In order to help a client discovering support for this profile, the directory object of an ACME server MUST contain the following attribute inside the "meta" field:
+
+- star-delegation-enabled: boolean flag indicating support for the profile specified in this memo.  An ACME server that supports this delegation profile MUST include this key, and MUST set it to true.
+
+### On Cancelation 
+
+It is worth noting that cancelation of the ACME STAR certificate is a prerogative of the IdO.  The NDC does not own the relevant account key on the ACME CA, therefore it can't issue a cancelation request for the STAR cert.  Potentially, since it holds the STAR cert private key, it could request the revocation of a single STAR certificate.  However, STAR explicitly disables the revokeCert interface.
+
+[[TBD.2: On key compromise, how does the NDC signal the IdO the need of cancelation?]]
+
+# CDNI Use Cases
+
+Members of the IETF CDNI (Content Delivery Network Interconnection) working group are interested in delegating authority over web content to CDNs.  Their requirements are described in a draft {{I-D.fieau-cdni-https-delegation}} that considers several solutions addressing different delegation requirements.  This section discusses two of these particular requirements in the context of the STAR delegation workflow.
+
+## Multiple Parallel Delegates
+
+In some cases the content owner (IdO) would like to delegate authority over a web site to multiple NDCs (CDNs).  This could happen if the IdO has agreements in place with different regional CDNs for different geographical regions, or if a "backup" CDN is used to handle overflow traffic by temporarily altering some of the CNAME mappings in place.  The STAR delegation flow enables this use case naturally, since each CDN can authenticate separately to the IdO specifying its CSR, and the IdO is free to allow or deny each certificate request according to its own policy.
```

Comment:
In particular, each CDN will have a separate account on the IdO.

Reply:
OK